### PR TITLE
fix(data-table): add a width to the expand icon container

### LINKED ISSUE #1260: Make expand-collapse column in Data Table fixed width, to avoid jiggle
### Summary

The Data Table component jiggles when you expand and collapse a row inside it, because the expand-collapse arrow changes width when it is rotated by 90 degrees. We recommend giving the expand-collapse-arrow column a fixed width, so the table doesn't jiggle when expanded. 

This is a design & dev item.

### Justification

Movement of the table when expanding a row draws unnecessary attention to the table itself, when the area the user is concerned with is the expanded state of the row.

### Extra stuff

This effect occurs on the carbon website (https://www.carbondesignsystem.com/components/data-table/code) as well as in production.

### PR DESCRIPTION
**Pull request by aledavila**
Closes #1260 

#### Changelog

**New**

- added width to icon expand container to prevent slight width change on rows when expanding. 

#### Testing / Reviewing

Check dev environment is not broken.
Check DataTableV2 expandable

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
LGTM 👍✅ - Thanks @aledavila!

**Comment by alisonjoseph** [comment]
I'm still seeing a jump in the width by 1px. Looking at what else might be causing it.

File: src/components/data-table-v2/_data-table-v2-expandable.scss [hunk 1]
```diff
@@ -69,6 +69,10 @@
     }
   }
 
+  .#{$prefix}--table-expand-v2 {
+    width: 2.25rem;
```
Comment (alisonjoseph) [inline src/components/data-table-v2/_data-table-v2-expandable.scss]: If you bump this to 2.5 it seems to fix it. 🙏

Comment (aledavila) [reply src/components/data-table-v2/_data-table-v2-expandable.scss]: I've updated it now

**Review by alisonjoseph — APPROVED** [review]

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**